API for single level Pie Chart.

himanshu-balasamanta opened the pull request:
User can change colour of element to his wish.
Legend has all elements of the series.
Produces perfectly circular chart.

Created a String-Array data model...(This can potentially be used set to use for category series.. )

If this is good, may it be merged.  Extending it to extensions bundle shall be done during GSoC period, during which time Multi-Level Pie-charts API will be added and together Single and multi-level Pie Charts will be added to Extensions bundle.
![PieChart](https://user-images.githubusercontent.com/44469499/78541275-044ba700-7813-11ea-9346-716233ee711a.png)

himanshu-balasamanta committed:
Single Level PieChart added with an example

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
components of plot area set.

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
Merge remote-tracking branch 'upstream/develop' into pie

himanshu-balasamanta committed:
minor mistakes undone

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
shifted data model related methods from Series to PieSeries

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
Legend set

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
added ability to set colors.

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
minor changes in pie chart example

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
set legend for allowing SWT.BOTTOM and SWT.TOP position

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
unnecessary spaces removed

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

eselmeister commented:
@Himanshu-Balasamanta The code needs improvements:

A) Error: the "Color[] color = series.getColors();" method can't be resolved in:

- org.eclipse.swtchart.internal.Legend

B) Please add appropriate header files, e.g.:

- org.eclipse.swtchart.internal.compress.CompressPieSeries
- org.eclipse.swtchart.internal.compress.IPieSeries
- org.eclipse.swtchart.internal.compress.PieSeries
- ...

C) Please format the source code before committing it. The styles are included already. You have to activate the "Save actions" in Eclipse to auto format the code on each change.

D) Class member shall be private. Some classes have members which are currently "package protected". Please use private members:

- String[] labels;
- double[] values;
- Color[] colors;

E) Methods: Please list protected methods after the public methods

F) When changing code, please update the copyright header, e.g.

- Copyright (c) 2008, 2019 SWTChart project. -> Copyright (c) 2008, 2020 SWTChart project.

E) StringArrayModel
Could it be used except for the pie series?
```
public void setStringArrayModel(StringArraySeriesModel data) {

	this.stringArraySeriesModel = data;
	setCompressor();
	if(compressor instanceof CompressPieSeries) {
		((CompressPieSeries)compressor).setLabelSeries(getLabelSeries());
		((CompressPieSeries)compressor).setValueSeries(getValueSeries());
	} else if(compressor instanceof CompressBarSeries) {
		// code for category series
	}
}
```

himanshu-balasamanta committed:
brought suggested changes

*source code has been formatted.
*class members accessibility is restricted to private.
*methods are now listed in the order public, then protected, then private
*copyright header is updated for all files that are changed.
*StringArrayModel, if used to set for Valid category BarSeries will be a messy affair.
*As of now, useing it for PieSeries only.

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

himanshu-balasamanta committed:
added boundaries to chart

Signed-off-by: Himanshu Balasamanta <himanshubb.eee18@itbhu.ac.in>

eselmeister commented:
@Himanshu-Balasamanta Please address all of my above listed issues. Some examples, which have not been adjusted. Please go through all your changes thoroughly, review them and format the source code.

![Bildschirmfoto von 2020-04-07 07-38-51](https://user-images.githubusercontent.com/686671/78634177-a9ae5b80-78a3-11ea-8320-19ebf8745c09.png)
![Bildschirmfoto von 2020-04-07 07-39-10](https://user-images.githubusercontent.com/686671/78634178-aadf8880-78a3-11ea-80fe-e3f6c78fa7c0.png)
![Bildschirmfoto von 2020-04-07 07-39-26](https://user-images.githubusercontent.com/686671/78634180-aadf8880-78a3-11ea-8744-ee208d6ec126.png)

himanshu-balasamanta commented:
I am very sorry sir, I am unable to figure out where the mistake is. 
IPieSeries is not empty
![Screenshot from 2020-04-07 11-52-09](https://user-images.githubusercontent.com/44469499/78639560-cf4f5b00-78cb-11ea-903b-6b6315819949.png)
pieSeries.getColors is fine.
![Screenshot from 2020-04-07 12-33-15](https://user-images.githubusercontent.com/44469499/78639712-0a518e80-78cc-11ea-891f-e282ff0fc67c.png)
attributes are private marked
![Screenshot from 2020-04-07 11-48-44](https://user-images.githubusercontent.com/44469499/78639786-33721f00-78cc-11ea-9418-b73620cf1faa.png)
I am not sure of what I am to do.

eselmeister commented:
@Himanshu-Balasamanta No worries, it's my job to mentor and guide you. I'll have a look at the issues and give you a detailed feedback soon.

Himanshu-Balasamanta commented:
Should I close and open the pull request again?
Very sorry for all the trouble.

eselmeister commented:
No worries. Leave it as it is. We will go through the issues together. I 
have to focus on another task now. I'll have a look at your issue in the 
afternoon.


Best,
Philip

Am 07.04.20 um 09:21 schrieb Himanshu Balasamanta:
>
> Should I close and open the pull request again?
> Very sorry for all the trouble.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/eclipse/swtchart/pull/138#issuecomment-610221175>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFHUT72V4BC2JKESTNVQKLRLLIA7ANCNFSM4MCCYDNQ>.
>

-- 
~~~~~~~~~~~~~~~~~~~~~~~~
OpenChrom - the open source alternative for chromatography / mass spectrometry
Dr. Philip Wenig » Founder » philip.wenig@openchrom.net » http://www.openchrom.net
~~~~~~~~~~~~~~~~~~~~~~~~
